### ISSUE TITLE
Export as File - Missing Index

### ISSUE THREAD

**Issue by robert-swan-as** [microsoft/vscode-data-wrangler#202]
Type: <b>Bug</b>

# Behaviour

## Expected vs. Actual

Using the export as file functionality (to csv) on a pandas dataframe does not export the index, only column data.

## Steps to reproduce:

import pandas as pd

d = {'col1': ['row1', 'row2'], 'col2': [3, 4], 'col3': [5, 6]}
df = pd.DataFrame(data=d)
df = df.set_index('col1')

Exporting "df" to csv only has data from col2 and col3

# Diagnostic data

-   Jupyter extension version: 2024.5.0
-   Python extension version: 2024.8.0
-   .NET Install Tool for Extension Authors extension version: Not installed
-   Python package dependencies:

```json
{
  "installed": {
    "pandas": "2.2.0"
  },
  "required": {
    "pandas": "0.25.2"
  },
  "unsatisfied": []
}
```

-   Entrypoint: JupyterDataViewer
-   Active mode: dataViewer

Extension version: 1.2.0
VS Code version: Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573, 2024-06-04T19:33:54.889Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 x 2803)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.73GB (12.29GB free)|
|Process Argv|--crash-reporter-id 25d70cdd-c2cf-4438-86d4-fb6cc02eb31b|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyonecf:30548226
vscrp:30673768
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
724cj586:31013169
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupyter:31046869
26j00206:31048877

```

</details>

**Comment by kycutler** [comment]
Hi @robert-swan-as, thanks for opening this issue!

We have updated the CSV export behavior to address this issue. Now, the index will be included if it is not the default index (i.e. `df.index.name` is not `None`). To try it out, please install the latest pre-release version of the Data Wrangler extension (1.3.1).

Closing this issue as it should be resolved, but feel free to re-open if it doesn't work for you.

Thanks for trying Data Wrangler!

**Comment by robert-swan-as** [comment]
Hi,

The same issue is occurring for Pandas Series objects. i.e the Index isn't being exported.

Thanks,

Robert

**Comment by arseru** [comment]
@kycutler I can confirm that what @robert-swan-as mentions happening for Series is happening to me as well - naming the index (i.e. `my_series.index.name = "foobar"`) seems to fix it, and the export to CSV gets corrected this way.

**Comment by billtubbs** [comment]
Has this fix been released?  My Data Wrangler version is "Last updated 2024-11-14, 01:00:36" but I'm not seeing the index in the exported csv file.  The index is named.  It's a mutli-index though so, index.name is a tuble. I'm wondering if that could be the reason.

**Comment by kycutler** [comment]
> Has this fix been released? My Data Wrangler version is "Last updated 2024-11-14, 01:00:36" but I'm not seeing the index in the exported csv file. The index is named. It's a mutli-index though so, index.name is a tuple. I'm wondering if that could be the reason.

Hi @billtubbs, thanks for reporting this!

It looks like for a `MultiIndex`, `index.name` may still return `None`, so we needed to test using `index.names[0]` instead. This should be fixed now in the latest pre-release version `1.15.1`. But please let me know if you still see the issue!

**Comment by kycutler** [comment]
Hi @ruimoreno, can you confirm whether the indexes are named (i.e. `df.index.names[0]` is not `None`)? If not, could you share specifics about how the index was set, so we can look into supporting your scenario?
